cartographer_ros/cartographer_ros/occupancy_grid.cc

```diff
@@ -58,31 +60,34 @@ Eigen::AlignedBox2f ComputeMapBoundingBox(
 namespace cartographer_ros {
 
 void BuildOccupancyGrid2D(
-    const std::vector<::cartographer::mapping::TrajectoryNode>&
-        trajectory_nodes,
+    const std::vector<std::vector<::cartographer::mapping::TrajectoryNode>>&
+        all_trajectory_nodes,
     const string& map_frame,
     const ::cartographer::mapping_2d::proto::SubmapsOptions& submaps_options,
     ::nav_msgs::OccupancyGrid* const occupancy_grid) {
-  CHECK(!trajectory_nodes.empty());
+  CHECK(!all_trajectory_nodes.empty());
```

Comment:
This now checks that there is a trajectory but we wanted to check if there is some node to compute proper map limits. Also, nodes can be trimmed now so that check would also no longer make sense. If there is no non-trimmed node, we will not have a proper time set. `CHECK` that latest_time is not `min()` anymore afterwards?